classes/local/datasource/roles.php

```diff
@@ -246,6 +246,18 @@ public function config_on_update($data) {
         $this->update_config_cache_key();
     }
 
+    /**
+     * Called when an instance is added/updated.
+     *
+     * @param \stdClass $instancedata
+     * @param bool $isnewinstance
+     */
+    public function update_instance(\stdClass $instancedata, bool $isnewinstance) {
+        $cacherev = $this->cms->get_custom_data('roles_course_role_cache_rev') ?? 0;
+        $this->cms->set_custom_data('roles_course_role_cache_rev', ++$cacherev);
```

Comment:
So does this mean the cacherev will always start at 1? Since if it's empty/does not exist it will return 0 but then we pre-increment when saving into the custom data

Reply:
That's right.

Reply:
I use the '??' here because it gets called when creating, and so there is not a rev number stored yet.